client/client.go

```diff
@@ -717,13 +785,98 @@ func newWebsocketDialerTLS(conf Config) (*websocket.Dialer, error) {
 
 	dialer := websocket.Dialer{
 		NetDialTLSContext: func(_ context.Context, network string, addr string) (net.Conn, error) {
+			proxyUrl, err := getProxyUrlFromHostPort(addr)
+			if err != nil {
+				log.Errorf("Failed to get http-proxy configurations during websocket dial: %v", err)
+			} else if proxyUrl != nil {
+				return proxyDialOpenSSL(ctx, &conf, network, addr, proxyUrl)
+			}
 			return dialOpenSSL(ctx, &conf, network, addr)
 		},
 	}
 
 	return &dialer, nil
 }
 
+func getProxyUrlFromHostPort(addr string) (*url.URL, error) {
+	u, err := url.Parse("https://" + addr)
```

Comment:
`http` proxy is not supported?

Reply:
Problem: 

The `Proxy` field in the `websocket.Dialer` and `http.Transport` is a function that takes an `*http.Request` and returns an `*url,URL`. Because it is a function it can be customized to dynamically decide if proxying should be used and what the proxy target is given a request. The request itself has an `*url.URL` which is what is used to decide the aforementioned points.
 
However, in `NetDialTLSContext` we don`t have the full url. Instead we have `addr` whos value is `URL.Host` (scheme is unknown). So what this does is to assume that the target is using `HTTPS`, which I think is OK since the mender-client won't talk to servers using unencrypted `HTTP` anyway.

Reply:
@lluiscampos 
A side note here: we don't really support the `HTTP_PROXY` variable. It implies that the `ServerURL`'s scheme is HTTP and we don't allow application layer communication without any form of encryption.

Reply:
Okey.